data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManager.java

```diff
@@ -48,6 +48,9 @@ public void setupIndex() throws IOException {
 
     @Override
     public String getIndexName(final String dynamicIndexAlias) throws IOException {
+        if (dynamicIndexAlias == null) {
+            throw new IOException("index alias is null");
```

Comment:
From what I can tell, this will still cause data prepper to crash when the alias is null. Is there a way we can validate that this is not null during initialization? Crashing on initialization rather than during ingestion would be a better user experience

Reply:
> Is there a way we can validate that this is not null during initialization?

It looks like this value is dependent on the events that are ingested. In that case, I don't think validation during initialization is possible. I am in agreement with David's comment.

Reply:
@dlvenable @engechas we can go either way. Using null string when the property is missing, vs discarding the event. If the config has `index-name: ${prop}` and the `prop` does not exist in the event, the index-name will become nullstring and it will be discarded anyways.

Reply:
I think we should not crash Data Prepper, any event at at some point can have null value. I think we should have a null string and log the error.

Reply:
Discarding the event seems reasonable as well. I think we should [log this as a warning here](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java#L142) with some message like "Dropping event due to missing field {}. Field {} is required for the configured DynamicIndexTemplate." and create a custom exception for this case rather than reusing IOException.

Reply:
@asifsmohammed @engechas I do not think we should log, because this would be logged for every event and there could be millions of events. I have discussed with Dinu and I think introducing a metrics counter is better.